sw/device/silicon_creator/lib/sigverify.c

```diff
@@ -72,17 +72,61 @@ static rom_error_t sigverify_padding_and_digest_check(
   return kErrorOk;
 }
 
+/**
+ * Determines whether the software implementation should be used for signature
+ * verification.
+ *
+ * During manufacturing (TEST_UNLOCKED*), software implementation is used by
+ * default since OTP may not have been programmed yet. The implementation to use
+ * after manufacturing (PROD, PROD_END, DEV, RMA) is determined by the OTP
+ * value.
+ *
+ * @param lc_state Life cycle state of the device.
+ * @param[out] use_sw Use software implementation for signature verification.
+ * @return Result of the operation.
+ */
+static rom_error_t sigverify_use_sw_rsa_verify(lifecycle_state_t lc_state,
+                                               hardened_bool_t *use_sw) {
+  switch (lc_state) {
+    case kLcStateTestUnlocked0:
+    case kLcStateTestUnlocked1:
+    case kLcStateTestUnlocked2:
+    case kLcStateTestUnlocked3:
+    case kLcStateTestUnlocked4:
+    case kLcStateTestUnlocked5:
+    case kLcStateTestUnlocked6:
+    case kLcStateTestUnlocked7:
```

Comment:
Might be worth a adding some inline functions in `lifecycle.h` to check for common combinations. I can see something like `lifecycle_is_otp_programmed(lifecycle_state_t)` or `lifecycle_is_test_unlocked(lifecycle_state_t)` being useful elsewhere too. This is fine as-is, just something to consider as we add more of these switch statements.

Reply:
Good point. I also thought about this when working on #7654 and this PR but decided to wait and see. Higher level functions like `is_test_unlocked`, `is_non_test_operational` would be useful if we want to treat all TEST_UNLOCKED* states the same. `is_otp_programmed` would also be useful but I guess depends on when we program the OTP during manufacturing and when we want to use those values.